multi/stm32l4-multi/flash.c

```diff
@@ -233,6 +259,54 @@ size_t flash_writeData(uint32_t offset, const char *buff, size_t size)
 }
 
 
+size_t flash_writeOtp(uint32_t offset, const char *buff, size_t size)
+{
+	size_t ret, len;
+	size_t pos = 0;
+	uint32_t temp[2];
+	volatile uint32_t *ptr = (void *)offset;
+
+	ret = otp_isValidAddress(offset, size);
+	if (!ret) {
+		return 0;
+	}
+
+	mutexLock(flash_common.lock);
+	_program_unlock();
+	_flash_clearFlags();
+
+	ret = _flash_wait();
+	if (ret == 0) {
+		for (pos = 0; pos < size; pos += OTP_UNIT_SIZE) {
+			*(flash_common.flash + flash_cr) |= 1;
+			len = pos + OTP_UNIT_SIZE > size ? size - pos : OTP_UNIT_SIZE;
+			if (len % OTP_UNIT_SIZE) {
+				memset(temp, 0xff, OTP_UNIT_SIZE);
+			}
```

Comment:
I think it would be safer to not perform any writes that are shorter than OTP_UNIT_SIZE. Data cannot be programmed again, even if we use only a part of OTP word.
E.g. some not-so-smart function on some level that performs write byte-by-byte - it will be only able to write every 8th byte successfully

Reply:
Also, you don't allow unaligned beginning, why allow unaligned end then?

Reply:
Now it is refusing not aligned address as well as not aligned data size.